lib/messages/status.js

```diff
@@ -4,15 +4,25 @@ const {
 } = require('.');
 
 class Status extends Message {
-  constructor(status) {
+  constructor(checkOrStatus) {
     super({});
-    this.status = status;
+    /*  error, failure, pending, success, queued, in_progress,
+        completed, neutral, cancelled, timed_out, or action_required
+    */
+    this.overallStatus = checkOrStatus.state || checkOrStatus.conclusion || checkOrStatus.status;
+    this.context = checkOrStatus.context || checkOrStatus.app.name || checkOrStatus.name;
```

Comment:
Would we ever get to that third conditional here?

We should always either have a `checkOrStatus.context` or `checkOrStatus.app.name`, right? 😅

Reply:
Actually, on second thought isn't `context` also always required? haha

Reply:
yeah that's probably true, I was going overboard to make sure we fixed #866